src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -15,12 +15,19 @@ internal class AzureKeyVaultSecretProvider
     {
         private readonly IDictionary<string, SecretClient> _secretClients;
         private readonly TokenCredential _credential;
+        private readonly Func<Uri, ValueTask<string>> _secretResolver;
 
-        public AzureKeyVaultSecretProvider(TokenCredential credential = null, IEnumerable<SecretClient> secretClients = null)
+        public AzureKeyVaultSecretProvider(TokenCredential credential = null, IEnumerable<SecretClient> secretClients = null, Func<Uri, ValueTask<string>> secretResolver = null)
         {
+            if (credential != null && secretResolver != null)
+            {
+                throw new InvalidOperationException($"Cannot configure both default credentials and secret resolver for Key Vault references. Please call only one of 'SetCredential()' or 'SetSecretResolver()' methods.");
+            }
```

Comment:
Could this check be moved earlier in the SetCredential and SetSecretResolver methods? Or alternatively in the ConfigureKeyVault method?

Reply:
My vote would be in the ConfigureKeyVault method. The earlier the better.